pallets/mixer/src/lib.rs

```diff
@@ -0,0 +1,251 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// A runtime module Groups with necessary imports
+
+/// Feel free to remove or edit this file as needed.
+/// If you change the name of this file, make sure to update its references in runtime/src/lib.rs
+/// If you remove this file, you can remove those references
+
+/// For more guidance on Substrate modules, see the example module
+/// https://github.com/paritytech/substrate/blob/master/frame/example/src/lib.rs
+
+#[cfg(test)]
+pub mod mock;
+
+#[cfg(test)]
+pub mod tests;
+
+use sp_runtime::traits::One;
+use sp_runtime::traits::AccountIdConversion;
+use sp_runtime::ModuleId;
+use merkle::merkle::keys::Commitment;
+use sp_runtime::traits::{Zero};
+use merkle::merkle::keys::Data;
+
+use frame_support::traits::{Currency, Get, ExistenceRequirement::{AllowDeath}};
+
+use codec::{Decode, Encode};
+use frame_support::{decl_error, decl_event, decl_module, decl_storage, dispatch, ensure};
+use frame_system::ensure_signed;
+use sp_std::prelude::*;
+use merkle::{Group as GroupTrait};
+
+pub type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+/// The pallet's configuration trait.
+pub trait Config: frame_system::Config + merkle::Config {
+	type ModuleId: Get<ModuleId>;
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+	/// Currency type for taking deposits
+	type Currency: Currency<Self::AccountId>;
+	/// The overarching group trait
+	type Group: GroupTrait<Self::AccountId, Self::BlockNumber, Self::GroupId>;
+	/// The max depth of the mixers
+	type MaxTreeDepth: Get<u8>;
+	/// The small deposit length
+	type DepositLength: Get<Self::BlockNumber>;
+}
+
+
+#[derive(Encode, Decode, PartialEq)]
+pub struct MixerInfo<T: Config> {
+	pub minimum_deposit_length_for_reward: T::BlockNumber,
+	pub fixed_deposit_size: BalanceOf<T>,
+	pub leaves: Vec<Data>,
+}
+
+impl<T: Config> core::default::Default for MixerInfo<T> {
+	fn default() -> Self {
+		Self {
+			minimum_deposit_length_for_reward: Zero::zero(),
+			fixed_deposit_size: Zero::zero(),
+			leaves: Vec::new(),
+		}
+	}
+} 
+
+
+impl<T: Config> MixerInfo<T> {
+	pub fn new(min_dep_length: T::BlockNumber, dep_size: BalanceOf<T>, leaves: Vec<Data>) -> Self {
+		Self {
+			minimum_deposit_length_for_reward: min_dep_length,
+			fixed_deposit_size: dep_size,
+			leaves: leaves,
+		}
+	}
+}
+
+// This pallet's storage items.
+decl_storage! {
+	trait Store for Module<T: Config> as Mixer {
+		pub Initialised get(fn initialised): bool;
+		/// The map of mixer groups to their metadata
+		pub MixerGroups get(fn mixer_groups): map hasher(blake2_128_concat) T::GroupId => MixerInfo<T>;
+		/// Map of used nullifiers (Data) for each tree.
+		pub UsedNullifiers get(fn used_nullifiers): map hasher(blake2_128_concat) (T::GroupId, Data) => bool;
```

Comment:
This is not used

Reply:
Good catch, this is fixed and ya was handled in Group pallet.